.github/workflows/test-go.yaml

```diff
@@ -0,0 +1,54 @@
+name: Go Test
+
+on:
+  - push
+  - pull_request
+
+jobs:
+  test:
+    name: Test
+    runs-on: ubuntu-latest
+    env:
+      GOPATH: ${{ github.workspace }}/go
+    defaults:
+      run:
+        working-directory: ${{ env.GOPATH }}/src/github.com/kubeflow/training-operator
+
+    steps:
+      - name: Check out code
+        uses: actions/checkout@v2
+        with:
+          path: ${{ env.GOPATH }}/src/github.com/kubeflow/training-operator
+
+      - name: Setup Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: 1.14
+
+      - name: Check Go modules
+        run: |
+          go mod tidy && git add go.*
+          git diff --cached --exit-code || (echo "Please run "go mod tidy" to sync Go modules" && exit1);
+      - name: Check manifests
+        run: |
+          make manifests && git add manifests
+          git diff --cached --exit-code || (echo "Please run "make manifests" to generate manifests" && exit1);
+      - name: Check auto-generated codes
+        run: |
+          make generate && git add pkg/apis
+          git diff --cached --exit-code || (echo "Please run "make generate" to generate Go codes" && exit1);
+      - name: Verify gofmt
+        run: |
+          make fmt && git add pkg cmd
+          git diff --cached --exit-code || (echo "Please run "make fmt" to verify gofmt" && exit1);
+      - name: Verify govet
+        run: |
+          make vet && git add pkg cmd
+          git diff --cached --exit-code || (echo "Please run "make vet" to verify govet" && exit1);
+      - name: Run Go test
+        run: make test
+      - name: Coveralls report
```

Comment:
@tenzen-y @Jeffwan Do you know is it possible to use separate GitHub action for the Coveralls report ?
Similar to this PR check list: https://github.com/coverallsapp/coveralls-node-demo/pull/18.
`coverage/coveralls — Coverage remained the same at 92.0%`

I think it will be very useful if contributors will see coverage % in the action list and redirect link.

Reply:
Technically, yes. It need a separate flow. I don't have strong opnions on worflows. both work for me

Reply:
@andreyvelich 
I'm not familiar with coveralls and don't immediately know how to display coverage as in  https://github.com/coverallsapp/coveralls-node-demo/pull/18, so I'll create a separate issue. Does it sound good?

Reply:
@andreyvelich @Jeffwan 
I created [this issue](https://github.com/kubeflow/training-operator/issues/1442).